kivy/uix/rst.py

```diff
@@ -645,9 +646,14 @@ def goto(self, ref, *largs):
         # search into all the nodes containing anchors
         ax = ay = None
         for node in self.anchors_widgets:
-            if ref in node.anchors:
-                ax, ay = node.anchors[ref]
-                break
+            try:
+                if ref in node.anchors:
+                    ax, ay = node.anchors[ref]
+                    break
+            except AttributeError:
```

Comment:
Maybe switching to something more precise such as [`hasattr`](https://docs.python.org/3.6/library/functions.html#hasattr) would be more efficient. It'll even make the code shorter (because `break`). Just a suggestion ^^